PsychSourceGL/Cohorts/FTGLTextRenderer/OGLFT.cpp

```diff
@@ -3442,6 +3494,22 @@ QString Face::format_number ( const QString& format, double number ) { return( Q
     
     glEnd();
 
+    if (do_draw_underline_)
+    {
+        float undPos   = underline_position();
+        float undThicc = underline_thickness();
+	float advance = face->glyph->advance.x/64.f;
+	
+	glBindTexture( GL_TEXTURE_2D, 0);
```

Comment:
Nope
No glBindTeture(..,0) or glColor3F()

Rather glDisable(GL_TEXTURE_2D); and possibly a matching glEnable(GL_TEXTURE_2D) at the end of the branch.

This would disable application of the glyph texture, and i think the vertex color will already be set to the text color and therefore the underline color. Unless you want all text always underlined in black, which doesn't make sense?

Reply:
Yeah, that wouldn't be right if underlines are always black, will fix as suggested